backend/src/main/java/com/beaplab/BeaplabEngine/controller/WatchController.java

```diff
@@ -0,0 +1,295 @@
+        // getting the appropriate viewName
+        String viewName = "";
+        switch (watchType) {
+            case "applewatch":
+                viewName = "appleWatch";
+                break;
+            case "fitbit":
+                viewName = "fitbit";
+                break;
+            default:
+                rollbar.error("Incorrect watch type selected: " + watchType);
+                return null;
+        }
+
+        ModelAndView model = new ModelAndView(viewName + "FileUpload");
+        return model;
+    }
+
+    // upload
+    @RequestMapping(value = "/{watchType}/upload", method = RequestMethod.POST)
+    @ApiOperation(value = "Zip file upload", notes = "Uploading a zip file containing Apple Watch raw data and persisting it to database")
+    @ApiResponses(value = {
+            @ApiResponse(code = 415, message = "{success: false, message: Invalid file type}"),
+            @ApiResponse(code = 200, message = "{success: true}, raw_data_id: XXXX}"),
+            @ApiResponse(code = 400, message = "{success: false, message: Invalid watch type}"),
+            @ApiResponse(code = 406, message = "{success: false, message: One zip file is acceptable}"),
+            @ApiResponse(code = 500, message = "{success: false, message: XYZ}")
+    })
+    @ApiImplicitParams({
+            @ApiImplicitParam(name = "file", value = "the zipped file to be uploaded as multipart/form-data", type = "POST", required = true, paramType = "blob")
+    })
+    public ResponseEntity<JSONObject> upload(
+            @ApiParam(value = "The watch type to be uploaded", required = true) @PathVariable("watchType") String watchType,
+            @ApiIgnore MultipartHttpServletRequest request) {
+
+        rollbar.info("in WatchController/upload");
+
+        HttpSession session = request.getSession(false);
+        if (session == null || !request.isRequestedSessionIdValid()) { // invalid session
+            JSONObject jsonObject = new JSONObject();
+            jsonObject.put(BeapEngineConstants.SUCCESS_STR, false);
+            jsonObject.put("message", "Invalid session");
+            jsonObject.put("status_code", HttpStatus.NON_AUTHORITATIVE_INFORMATION.value());
+
+            return new ResponseEntity<>(jsonObject, (HttpStatus.valueOf((int) jsonObject.get("status_code"))));
+        }
+
+        if (!watchType.equals("applewatch") && !watchType.equals("fitbit")) {
+            JSONObject jsonObject = new JSONObject();
+            jsonObject.put(BeapEngineConstants.SUCCESS_STR, false);
+            jsonObject.put("message", "Invalid watch type in url");
+            jsonObject.put("status_code", 400);
+            return new ResponseEntity<>(jsonObject, (HttpStatus.valueOf((int) jsonObject.get("status_code"))));
+        }
+
+        SessionDetails sessionDetails = (SessionDetails) session.getAttribute("SESSION_DETAILS");
+
+        JSONObject result = new JSONObject();
+        switch (watchType) {
+            case "applewatch":
```

Comment:
put these strings inside variable also

Reply:
Switches in java only work with literals :(

I tried using variables (and I probably could switch to them) but it didn't work.
It'd have to be with a chain of if-else statements instead of a switch though